### ISSUE TITLE
Immutable tuple URL sqlachemy

### ISSUE THREAD

**Issue by DeclercqM** [miguelgrinberg/Flask-Migrate#388]
In the following [commit](https://github.com/sqlalchemy/sqlalchemy/commit/3668b3a30c4b9a9f4af13457f43bfc34c28cf448) has sqlalchemy decided to make it a tuple.

When executing `flask db migration` is this the result. 
```
  File "C:\Users\server\venv\lib\site-packages\flask_sqlalchemy-2.4.4-py3.6.egg\flask_sqlalchemy\__init__.py", line 944, in engine
    return self.get_engine()
  File "C:\Users\server\venv\lib\site-packages\flask_sqlalchemy-2.4.4-py3.6.egg\flask_sqlalchemy\__init__.py", line 963, in get_engine
    return connector.get_engine()
  File "C:\Users\server\venv\lib\site-packages\flask_sqlalchemy-2.4.4-py3.6.egg\flask_sqlalchemy\__init__.py", line 555, in get_engine
    options = self.get_options(sa_url, echo)
  File "C:\Users\server\venv\lib\site-packages\flask_sqlalchemy-2.4.4-py3.6.egg\flask_sqlalchemy\__init__.py", line 570, in get_options
    self._sa.apply_driver_hacks(self._app, sa_url, options)
  File "C:\Users\server\venv\lib\site-packages\flask_sqlalchemy-2.4.4-py3.6.egg\flask_sqlalchemy\__init__.py", line 915, in apply_driver_hacks
    sa_url.database = os.path.join(app.root_path, sa_url.database)
AttributeError: can't set attribute
```

**Comment by miguelgrinberg** [comment]
This is a Flask-SQLAlchemy error, it does not appear to be related to this package. Does your application work? If you think it is this package, then please provide an example application that I can use to reproduce the error.

**Comment by DeclercqM** [comment]
My apologies for posting this issue on the wrong package. On this [link](https://github.com/DeclercqM/example_application) you can find an example application to re produce the error. After installing the venv and the setup.py the next command to reproduce the error is `flask db init` and `flask migrate -m "message migration"`

**Comment by miguelgrinberg** [comment]
@DeclercqM Going back to my question, I asked if the application runs fine. I installed your repo and can't do `flask run`, I get weird errors. Your `__init__.py` file has an `from app import routes` yet there is no `app` package. There is nothing here that points to errors in this package, I think you need to review your basic application structure and make sure all the pieces are working well together.

**Comment by baezfb** [comment]
@DeclercqM I was getting the same error, I fixed it by changing SQLAlchemy version to 1.3.18.

**Comment by mukeshsharma1201** [comment]
Related Issue on flask-sqlalchemy : https://github.com/pallets/flask-sqlalchemy/issues/910
